rust/otap-dataflow/crates/otap/src/attributes_processor.rs

```diff
@@ -194,13 +194,15 @@ impl local::Processor<OtapPdata> for AttributesProcessor {
                 }
 
                 let signal = pdata.signal_type();
-                let mut records: OtapArrowRecords = pdata.try_into()?;
+                let (context, payload) = pdata.split();
```

Comment:
@jmacd , @albertlockett At some point, I'd like us to get to something as simple as this, which in my opinion is much simpler without sacrificing performance.

```rust
// If transfor_attributes do nothing then it returns self, otherwise it returns
//  a new pdata (including the context of consumed one).
let pdata = pdata.transform_attributes(&transforms);
effect_handler.send_message(pdata);
```